### ISSUE TITLE
preprocess of dataset pamap2

### ISSUE THREAD

**Issue by liujun486597** [saif-mahmud/self-attention-HAR#3]
https://github.com/saif-mahmud/self-attention-HAR/blob/a928840b63241ec1743b6a9419f88ab85f0f11d0/preprocess/pamap2/_data_reader.py#L78

maybe it should be `data.append([float(x) / 1000 for x in elem[1:]])`?

**Comment by XILOUYUE7** [comment]
I think so. maybe it should be data.append([float(x) / 1000 for x in elem[1:]])?

**Comment by rlleshi** [comment]
Because `elem[0]` represents the label, which we don't want to use as part of the input dataset, while `elem[-1]` represents the last gyro data for the IMU ankle, which we want to use as part of the input dataset.

**Comment by XILOUYUE7** [comment]
Thank you  for your  response.  After the preprocessing of pamap2 dataset, there are  19 columns of data left, elem[0] is the label, the other columns is data.  While as shown in  your code , data.append([float(x) / 1000 for x in elem[:-1]]) , dose it mean  processing the first 18 columns as  the data?

**Comment by rlleshi** [comment]
It looks like that if you change the line to `data.append([float(x) / 1000 for x in elem[1:]])`, the accuracy drops sharply. 

But it doesn't make sense to include the label itself to the training set. @saif-mahmud @thTonmoy could you provide some insights on this please?

**Comment by thTonmoy** [comment]
This project was a reimplementation for the official code using tensorflow 2 and is no longer actively maintained. There is probably some sort of issue with the preprocessing code which might need to be looked into. We hope to fix the potential issues when time permits.
